### PR TITLE
UX updates to user lists, added smart counts and bug fixes

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Adds:
* Total user counts for paging and searching user lists

Fixes:
* Spamming of next button in user lists
* Some UX issues with searching user lists (auto clear search, added button, etc.)
* Some hard to find/notice bugs

Removes:
* filtered_user_list.jsx now that it's not used anywhere

### CODE REVIEW

File: webapp/components/searchable_user_list.jsx [hunk 1]
```diff
@@ -30,64 +39,112 @@ export default class SearchableUserList extends React.Component {
...
 
+    componentWillUnmount() {
+        clearTimeout(this.nextTimeoutId);
+    }
+
     nextPage(e) {
         e.preventDefault();
         this.props.nextPage(this.state.page + 1);
-        this.setState({page: this.state.page + 1});
+        this.setState({page: this.state.page + 1, nextDisabled: true});
```
Comment (crspeller) [inline webapp/components/searchable_user_list.jsx]: this.props.nextPage could take some time and be async. Maybe set state before calling?

**Comment by jwilander** [comment]
Yep, good call

File: webapp/components/searchable_user_list.jsx (hunk 1 above)
Comment (jwilander) [inline webapp/components/searchable_user_list.jsx]: Yep, good call

Comment (jwilander) [inline webapp/components/searchable_user_list.jsx]: Fixed.

File: webapp/components/channel_invite_modal.jsx [hunk 2]
```diff
@@ -30,8 +30,8 @@ export default class ChannelInviteModal extends React.Component {
         this.handleInviteError = this.handleInviteError.bind(this);
         this.search = this.search.bind(this);
 
-        this.page = 0;
         this.term = '';
+        this.page = 0;
```
Comment (hmhealey) [inline webapp/components/channel_invite_modal.jsx]: Shouldn't term and page be part of the state? I've missed most of these initial changes so there's probably something I'm missing here

File: webapp/components/channel_members_modal.jsx [hunk 3]
```diff
@@ -34,6 +34,7 @@ export default class ChannelMembersModal extends React.Component {
         this.search = this.search.bind(this);
 
         this.term = '';
+        this.page = 0;
```
Comment (hmhealey) [inline webapp/components/channel_members_modal.jsx]: Same as my other comment

File: webapp/components/searchable_user_list.jsx [hunk 4]
```diff
@@ -18,9 +21,15 @@ export default class SearchableUserList extends React.Component {
...
         this.previousPage = this.previousPage.bind(this);
         this.doSearch = this.doSearch.bind(this);
+        this.onSearchBoxKeyPress = this.onSearchBoxKeyPress.bind(this);
+        this.onSearchBoxChange = this.onSearchBoxChange.bind(this);
+
+        this.nextTimeoutId = 0;
 
         this.state = {
-            page: 0
+            page: 0,
```
Comment (hmhealey) [inline webapp/components/searchable_user_list.jsx]: If we always have the page in the parent component, couldn't it just be passed in as a prop so we only need to track it in one place?

Comment (jwilander) [inline webapp/components/searchable_user_list.jsx]: We don't always have it in the parent, only in special circumstances

File: webapp/components/channel_invite_modal.jsx (hunk 2 above)
Comment (jwilander) [inline webapp/components/channel_invite_modal.jsx]: They're not used in the rendering, they're just workarounds. I wanted to keep as much of that logic in the lower component as possible but some cases like this one needed the information above.

**Pull request merged**